sdk/rust/oak/src/storage/mod.rs

```diff
@@ -88,19 +78,37 @@ impl Storage {
         let mut grpc_request = GrpcRequest::new();
         grpc_request.method_name = grpc_method_name.to_owned();
         grpc_request.set_req_msg(request_any);
+        let mut grpc_data = Vec::new();
+        grpc_request.write_to_writer(&mut grpc_data).unwrap();
+
+        // Create an ephemeral channel for the response.
+        let (rsp_out, rsp_in) = match crate::channel_create() {
+            Ok(x) => x,
+            Err(status) => {
+                return Err(grpc::build_status(
+                    grpc::Code::INTERNAL,
+                    &format!(
+                        "failed to create storage response channel: {}",
+                        status.value()
+                    ),
+                ))
+            }
+        };
 
-        grpc_request
-            .write_to_writer(&mut self.write_channel)
-            .unwrap();
+        crate::channel_write(self.write_channel, &grpc_data, &[rsp_out.handle]);
+        crate::channel_close(rsp_out.handle);
```

Comment:
This does not seem very intuitive, maybe we should think some more about linearity of handles.

Reply:
Yeah, there's a TODO in the Runtime planning doc.